docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,293 @@
+
+The Factory data set is kept in JSON format that is regulated by [JSON Schema](https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/nrfconnect/nrfconnect_factory_data.schema) file. All parameters are divided between `mandatory `and `optional.` 
+- `Mandatory` parameters must be provided because, without them, the device will not be able to perform commissioning to the Matter network. 
+- `Optional` parameters can be used for development and testing purposes. A `user` data parameter can consist of all data needed by a specific manufacturer that is not included in mandatory parameters.
+
+In the factory data set, there are some of the formats used. 
+
+- `uint16` and `uint32` are the numeric formats representing respectively two-bytes length unsigned integer and four-bytes length unsigned integer.
+- `hex string` meaning the bytes in hexadecimal representation given in a string. An prefix `hex:` is added to parameter to mark that is a `hex string`. E.g. an ASCII string *abba* is represented as *hex:61626261*, and its length is two times greater than ASCII string plus length of prefix. 
+- `ASCII string` is a string representation in ASCII encoding.
+- `ISO 8601` format is a [date format](https://www.iso.org/iso-8601-date-and-time-format.html) which represents a date given as YYYY-MM-DD or YYYYMMDD.
+- All certificates stored in factory data are given in [X.509](https://www.itu.int/rec/T-REC-X.509-201910-I/en) format.
+
+<hr>
+<a name="Using own factory data"></a>
+
+## Using own factory data
+
+
+
+<hr>
+<a name="Generating factory data"></a>
+
+## Generating factory data
+
+### Creating factory data JSON file
+
+### Verifying using a JSON schema
+
+Ready JSON file containing factory data can be verified using prepared [JSON Schema](https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/nrfconnect/nrfconnect_factory_data.schema). JSON Schema is a tool for validating the structure of JSON data. For this purpose, checking the correctness of given factory data is helpful. First, JSON Schema contains a list of mandatory parameters that must be stored in factory data. Missing at least one mandatory parameter will cause a checking error. Then, each parameter is checked according to the given pattern in the second phase of Schema work. The process of validating factory data JSON files is optional, but it can help avoid problems with reading parameters by Matter Device so it is strongly recommended to verify a newly created JSON file.
+
+To check JSON file using JSON Schema manually on Linux machine install *php-json-schema* package:
+```
+$ sudo apt install php-json-schema
+```
+
+And use following command:
+```
+$ validate-json <path_to_JSON_file> <path_to_schema_file>
+```
+
+If in the console there are not any outputs, it means that JSON is correct.
+
+JSON file can be checked automatically by the python script during its generation by given a path to JSON schema file as an additional argument:
```

Comment:
Hmm, so why we are recommending above 3rd party tool?

Reply:
I guess, in some cases, customers may don't want to use our python script to generate JSON but only use a partition creator. So I tried to point out that it can be done in some ways. Do you think we should force here validating JSON only by our script?